pkg/segment/writer/segstore.go

```diff
@@ -1024,6 +1023,52 @@ func (segstore *SegStore) WritePackedRecord(rawJson []byte, ts_millis uint64,
 	return nil
 }
 
+// This function will write bloom to file and return the number of bytes written
+// Ensure that the caller has to again write the file with the correct bloom size later.
+func (ss *SegStore) writeBloomToFile(fname string, bi *BloomIndex) (uint32, error) {
```

Comment:
The only use of this function was to prevent duplicacy for fd.Close(). Now we are directly opening the file in WRITE mode so no point in making this function. Addressed in the latest commit.